code/obj/item/clothing/chameleon_js.dm

```diff
@@ -269,3 +272,1557 @@
+		backpack_type = new/datum/chameleon_backpack_pattern/medic
+
+	roboticist
+		name = "Roboticist"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/roboticist
+		hat_type = new/datum/chameleon_hat_pattern
+		suit_type = new/datum/chameleon_suit_pattern/labcoat_robotics
+		glasses_type = new/datum/chameleon_glasses_pattern/prodoc
+		shoes_type = new/datum/chameleon_shoes_pattern
+		gloves_type = new/datum/chameleon_gloves_pattern/latex
+		belt_type = new/datum/chameleon_belt_pattern
+		backpack_type = new/datum/chameleon_backpack_pattern/robotics
+
+	geneticist
+		name = "Geneticist"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/geneticist
+		hat_type = new/datum/chameleon_hat_pattern
+		suit_type = new/datum/chameleon_suit_pattern/labcoat_genetics
+		glasses_type = new/datum/chameleon_glasses_pattern/prodoc
+		shoes_type = new/datum/chameleon_shoes_pattern/white
+		gloves_type = new/datum/chameleon_gloves_pattern/latex
+		belt_type = new/datum/chameleon_belt_pattern/medical
+		backpack_type = new/datum/chameleon_backpack_pattern/genetics
+
+	quartermaster
+		name = "Quartermaster"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/cargo
+		hat_type = new/datum/chameleon_hat_pattern
+		suit_type = new/datum/chameleon_suit_pattern
+		glasses_type = new/datum/chameleon_glasses_pattern
+		shoes_type = new/datum/chameleon_shoes_pattern
+		gloves_type = new/datum/chameleon_gloves_pattern
+		belt_type = new/datum/chameleon_belt_pattern
+		backpack_type = new/datum/chameleon_backpack_pattern
+
+	engineer
+		name = "Engineer"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/engineer
+		hat_type = new/datum/chameleon_hat_pattern/hardhat
+		suit_type = new/datum/chameleon_suit_pattern/winter_coat_engineering
+		glasses_type = new/datum/chameleon_glasses_pattern/meson
+		shoes_type = new/datum/chameleon_shoes_pattern/orange
+		gloves_type = new/datum/chameleon_gloves_pattern/insulated
+		belt_type = new/datum/chameleon_belt_pattern
+		backpack_type = new/datum/chameleon_backpack_pattern/engineer
+
+	mechanic
+		name = "Mechanic"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/mechanic
+		hat_type = new/datum/chameleon_hat_pattern/hardhat
+		suit_type = new/datum/chameleon_suit_pattern/winter_coat_engineering
+		glasses_type = new/datum/chameleon_glasses_pattern/meson
+		shoes_type = new/datum/chameleon_shoes_pattern/brown
+		gloves_type = new/datum/chameleon_gloves_pattern/insulated
+		belt_type = new/datum/chameleon_belt_pattern
+		backpack_type = new/datum/chameleon_backpack_pattern/engineer
+
+	miner
+		name = "Miner"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/overalls
+		hat_type = new/datum/chameleon_hat_pattern/space_helmet_engineer
+		suit_type = new/datum/chameleon_suit_pattern/space_suit_engineering
+		glasses_type = new/datum/chameleon_glasses_pattern/meson
+		shoes_type = new/datum/chameleon_shoes_pattern/orange
+		gloves_type = new/datum/chameleon_gloves_pattern
```

Comment:
I think this is wrong, since miners spawn with black gloves on their person but may not have had enough time in the round to create concussive gauntlets. That said, added the full sets of indy armor as options, so you can manually set them to be gauntlets if you wanted.